Formula/oauth2l.rb

```diff
@@ -0,0 +1,20 @@
+class Oauth2l < Formula
+  desc "Simple CLI for interacting with Google oauth tokens"
+  homepage "https://github.com/google/oauth2l"
+  url "https://github.com/google/oauth2l/archive/v1.0.1.tar.gz"
+  sha256 "8207bb796727019b277f63c709fa4e399f2d40c223633d0df6c74dc1f3c9385f"
+  head "https://github.com/google/oauth2l.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GO111MODULE"] = "on"
+
+    system "go", "build", "-o", "oauth2l"
+    bin.install "oauth2l"
+  end
+
+  test do
+    system "#{bin}/oauth2l", "-h"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Hi @SMillerDev - I added a better test. It would be good to get this added as a cop during audit.